vint.go

```diff
@@ -60,3 +60,46 @@ func mustReadByte(r io.Reader) (byte, error) {
 
 	return b, err
 }
+
+func WriteVInt(w io.Writer, n int64) (error) {
+	if n >= -112 && n <= 127 {
+		_, err := w.Write([]byte{byte(int8(n))})
+		return err
+	}
+
+	len := -112
+	if n < 0 {
+		n = n ^ -1
+		len = -120
+	}
+
+	tmp := n
+	for tmp != 0 {
+		tmp = tmp >> 8
+		len--
+	}
+
+	_, err := w.Write([]byte{byte(int8(len))})
+	if err != nil {
+		return err
+	}
+
+	if len < -120 {
+		len = -(len + 120)
+	} else {
+		len = -(len + 112)
+	}
+
+	for idx := len; idx != 0; idx-- {
+		shiftbits := uint((idx - 1) * 8)
+		mask := int64(0xFF << shiftbits)
+		masked := n & mask
+		currentbyte := masked >> shiftbits
+		_, err := w.Write([]byte{byte(currentbyte)})
```

Comment:
yeah I think this is basically

```go
for i := 1; i <= idx; i++ {
    b[i] = n >> (8 * (len - i))
}
```

Modulo off-by-one and endianness errors. =)